Kernel/Arch/aarch64/ASM_wrapper.h

```diff
@@ -64,11 +64,11 @@ inline void wait_cycles(int n)
 {
     // This is probably too fast when caching and branch prediction is turned on.
     // FIXME: Make timer-based.
-    asm("mov x0, %[value]\n"
+    asm("mov w0, %w[value]\n"
         "0:\n"
-        "    subs x0, x0, #1\n"
+        "    subs w0, w0, #1\n"
         "    bne 0b" ::[value] "r"(n)
-        : "x0");
+        : "w0");
```

Comment:
Even better, you can declare a scratch variable and take it as an in-out, to let the compiler decide which register to use
(might be even possible to write `r` as a clobber and referencing that, but haven't tested)

Reply:
x0 is general purpose in ARM ABIs so idk whether that matters...